### PR TITLE
docs: clarify analysis. Fixes #1234

### PR DESCRIPTION
**Pull request by kostis-codefresh**
Signed-off-by: Kostis Kapelonis <kostis@codefresh.io>

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [X] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [X] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -36,13 +36,15 @@ This is the mechanism that traffic from live users enters your cluster and is re
...
 
-## Analysis and AnalysisRun
+## AnalysisTemplate and AnalysisRun
 
-Analysis is a custom Kubernetes resource that connects a Rollout to your metrics provider and defines specific thresholds for certain metrics that will decide if a rollout is successful or not. For each Analysis you can define one or more metric queries along with their expected results. A Rollout will progress on its own if metric queries are good, rollback automatically if metrics show failure and pause the rollout if metrics cannot provide a success/failure answer.
+Analysis is the capability to connect a Rollout to your metrics provider and define specific thresholds for certain metrics that will decide if a rollout is successful or not. For each analysis you can define one or more metric queries along with their expected results. A Rollout will progress on its own if metric queries are good, rollback automatically if metrics show failure and pause the rollout if metrics cannot provide a success/failure answer.
```
Comment (huikang) [inline docs/architecture.md:41]: - How about ".... a rollout is successful or not" -> ".... an update is successful or not"?

File: docs/architecture.md [hunk 2]
```diff
@@ -36,13 +36,15 @@ This is the mechanism that traffic from live users enters your cluster and is re
...
 
-Analysis is just a template on what metrics to query. The actual result that is attached to a Rollout is the `AnalysisRun` custom resource. You can define an Analysis on a specific Rollout or globally on the cluster to be shared by multiple rollouts. The AnalysisRun resources is scoped on a specific rollout.
+For performing an analysis, Argo Rollouts includes two custom Kubernetes resources: `AnalysisTemplate` and `AnalysisResult`.
 
-Note that using Analysis and metrics in a Rollout is completely optional. You can manually pause and promote a rollout or use other external methods (e.g. smoke tests) via the API or the CLI. You don't need a metric solution just to use Argo Rollouts. You can also mix both automated (i.e. analysis based) and manual steps in a Rollout.
+`AnalysisTemplate` contains instructions on what metrics to query. The actual result that is attached to a Rollout is the `AnalysisRun` custom resource. You can define an `AnalysisTemplate` on a specific Rollout or globally on the cluster to be shared by multiple rollouts. The `AnalysisRun` resource is scoped on a specific rollout.
```
Comment (huikang) [inline docs/architecture.md:45]: Shall we mention `ClusterAnalysisTemplate` in this paragraph?

**Review by huikang — COMMENTED** [review]
some minor comments. Thanks for clarifying the concept of analysis.

File: docs/architecture.md (hunk 1 above)
Comment (kostis-codefresh) [reply docs/architecture.md]: @huikang done!

File: docs/architecture.md (hunk 2 above)
Comment (kostis-codefresh) [reply docs/architecture.md]: @huikang you are right. I added it.

**Review by huikang — APPROVED** [review]
LGTM

File: docs/architecture.md [hunk 3]
```diff
@@ -36,13 +36,15 @@ This is the mechanism that traffic from live users enters your cluster and is re
...
+Analysis is the capability to connect a Rollout to your metrics provider and define specific thresholds for certain metrics that will decide if an update is successful or not. For each analysis you can define one or more metric queries along with their expected results. A Rollout will progress on its own if metric queries are good, rollback automatically if metrics show failure and pause the rollout if metrics cannot provide a success/failure answer.
 
-Analysis is just a template on what metrics to query. The actual result that is attached to a Rollout is the `AnalysisRun` custom resource. You can define an Analysis on a specific Rollout or globally on the cluster to be shared by multiple rollouts. The AnalysisRun resources is scoped on a specific rollout.
+For performing an analysis, Argo Rollouts includes two custom Kubernetes resources: `AnalysisTemplate` and `AnalysisResult`.
```
Comment (jessesuen) [inline docs/architecture.md:43]: s/AnalysisResult/AnalysisRun/

**Review by jessesuen — CHANGES_REQUESTED** [review]
One typo, otherwise LGTM!

File: docs/architecture.md (hunk 3 above)
Comment (kostis-codefresh) [reply docs/architecture.md]: @jessesuen done

**Review by jessesuen — APPROVED** [review]

**Pull request merged**